taraxa/state/dpos/contract.go

```diff
@@ -375,10 +376,18 @@ func (self *Contract) upd_staking_balance(beneficiary common.Address, delta *big
 	new_vote_count := vote_count(beneficiary_bal, self.cfg.EligibilityBalanceThreshold, self.cfg.VoteEligibilityBalanceStep)
 	if prev_vote_count != new_vote_count {
 		self.eligible_vote_count -= prev_vote_count
-		self.eligible_vote_count += new_vote_count
+		self.eligible_vote_count = Add64p(self.eligible_vote_count, new_vote_count)
 	}
 }
 
+func Add64p(a, b uint64) uint64 {
+	c := a + b
+	if c < a || c < b {
+		panic("addition overflow " + strconv.FormatUint(a, 10) + " " + strconv.FormatUint(b, 10))
```

Comment:
I dont know, does not seem ok that our staking contract is be able to crash the node through panic on logic error like this. I think it could be handled in a nicer way:

upd_staking_balance function could return error similar to:
`func (self *Contract) apply_transfers(benefactor common.Address, transfers Transfers) (err error)`

That way, we can log the error in C++ part and node can continue without crash...

Reply:
no we are not logging it, we do panic so it is same
```	
if err := self.apply_transfers(entry.Benefactor, transfers); err != nil {
			panic(err)
		}
```
Anyway, this is not recoverable error. When this happens you should not be able to run the node anymore